autotest/gdrivers/saga.py

```diff
@@ -191,5 +191,37 @@ def test_saga_8():
     PARAMETER["false_northing",0],
     UNIT["Meter",1]]""")
 
+##############################################################################
+# Test setnodata
+def test_saga_9():
+
+    gdal_type = gdal.GDT_Float64
+
+    ds = gdal.GetDriverByName('SAGA').Create('tmp/test9.sdat', 2, 2, 1, gdal_type)
+    ds = None
+
+    ds = gdal.Open('tmp/test9.sdat')
+    ret = ds.GetRasterBand(1).SetNoDataValue(56)
```

Comment:
you may want to run that call under so that when running pytest in verbose mode (--capture=no -ra -vv), no error message is logged
```suggestion
    with gdaltest.error_handler():
        ret = ds.GetRasterBand(1).SetNoDataValue(56)
```